File: src/main/java/cn/afeibaili/mchat/socket/message/MessageType.java
package cn.afeibaili.mchat.socket.message;

/**
 * 消息类型
 *
 * @author AfeiBaili
 * @version 2026/3/3 00:36
 */

public abstract class MessageType {
    abstract String getIdentifier();

    abstract String getMessage();

    char delimiter = ':';

    @Override
    public String toString() {
        return getIdentifier() + delimiter + getMessage();
    }

    static class Text extends MessageType {
        String message;

        public Text(String message) {
            this.message = message;
        }

        @Override
        String getIdentifier() {
            return "txt";
        }

        @Override
        String getMessage() {
            return message;
        }
    }

    static class Command extends MessageType {
        String message;

        public Command(String message) {
            this.message = message;
        }

        @Override
        String getIdentifier() {
            return "cmd";
        }

        @Override
        String getMessage() {
            return message;
        }
    }

    static class Heartbeat extends MessageType {
        String message;

        public Heartbeat(String message) {
            this.message = message;
        }

        @Override
        String getIdentifier() {
            return "het";
        }

        @Override
        String getMessage() {
            return message;
        }
    }
}

File: core/src/main/kotlin/online/afeibaili/mchat/socket/message/MessageType.kt
package online.afeibaili.mchat.socket.message


/**
 * 消息类型
 *
 *@author AfeiBaili
 *@version 2025/11/3 18:39
 */

sealed class MessageType(val identifier: String) {
    abstract val message: String
    val delimiter: Char = ':'
    override fun toString(): String = "$identifier$delimiter$message"

    class Text(override val message: String) : MessageType("txt")
    class Command(override val message: String) : MessageType("cmd")
    class Heartbeat(override val message: String) : MessageType("het")
    class Image(override val message: String) : MessageType("img")
}